cartographer/mapping/map_builder.cc

```diff
@@ -52,8 +52,12 @@ proto::MapBuilderOptions CreateMapBuilderOptions(
   return options;
 }
 
-MapBuilder::MapBuilder(const proto::MapBuilderOptions& options)
-    : options_(options), thread_pool_(options.num_background_threads()) {
+MapBuilder::MapBuilder(
+    const proto::MapBuilderOptions& options,
+    std::unique_ptr<InsertedRangeDataCallback> inserted_range_data_callback)
```

Comment:
Why the `unique_ptr`? Seems unnecessary to me.

Reply:
The callback is optional, right? If you don't want to give one to libcartographer, you pass a null pointer during construction of `MapBuilder`.